timex/timex_test.go

```diff
@@ -0,0 +1,48 @@
+package timex
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+)
+
+func TestTicker(t *testing.T) {
+	ticker := NewTicker(2 * time.Millisecond)
+	time.AfterFunc(5*time.Millisecond, ticker.Stop)
+	cnt := 0
+	for range ticker.C {
+		cnt++
+	}
+	// Maybe flaky due to OS scheduling
+	require.Equal(t, 2, cnt)
+}
+
+func TestTickerPanic(t *testing.T) {
+	require.Panics(t, func() { NewTicker(-2 * time.Millisecond) })
+}
+
+func TestTickerWithContext(t *testing.T) {
+	errFive := fmt.Errorf("five")
+	cnt := 0
+	f := func() error {
+		cnt++
+		if cnt == 5 {
+			return errFive
+		}
+		return nil
+	}
+	g, ctx := errgroup.WithContext(context.Background())
```

Comment:
For those not familiar with errgroup, perhaps a comment saying "errgroup will cancel the context when one the functions it calls returns an error" or something. I had to look that up to understand the test.